pkg/common/db/cache/user.go

```
@@ -145,3 +157,70 @@ func (u *UserCacheRedis) DelUsersGlobalRecvMsgOpt(userIDs ...string) UserCache {
 	cache.AddKeys(keys...)
 	return cache
 }
+
+func (u *UserCacheRedis) getOnlineStatusKey(userID string) string {
+	return olineStatusKey + userID
+}
+
+// GetUserStatus get user status
+func (u *UserCacheRedis) GetUserStatus(ctx context.Context, userIDs []string) ([]*user.OnlineStatus, error) {
+	var res []*user.OnlineStatus
+	for _, userID := range userIDs {
+		UserIDNum, err := strconv.Atoi(userID)
+		if err != nil {
+			return nil, err
+		}
+		var modKey = strconv.Itoa(UserIDNum % statusMod)
+		var onlineStatus user.OnlineStatus
+		key := olineStatusKey + modKey
+		result, err := u.rdb.HGet(ctx, key, userID).Result()
+		if err != nil {
+			return nil, err
+		} else if err == redis.Nil {
+			// key or field does not exist
+			res = append(res, &user.OnlineStatus{
```

The error was not handled correctly